Public/Src/Cache/ContentStore/Interfaces/FileSystem/AbsFileSystemExtension.cs

```diff
@@ -17,6 +17,34 @@ public static class AbsFileSystemExtension
         /// </summary>
         public const int DefaultFileStreamBufferSize = 4 * 1024;
 
+        /// <summary>
+        ///     Try getting the attributes of a file.
+        /// </summary>
+        /// <returns>Returns false if the file doesn't exist.</returns>
+        public static bool TryGetFileAttributes(this IAbsFileSystem fileSystem, AbsolutePath path, out FileAttributes attributes)
+        {
+            if (!fileSystem.FileExists(path))
+            {
+                attributes = default;
+                return false;
+            }
+
+            try
+            {
+                attributes = fileSystem.GetFileAttributes(path);
+            }
+            catch(FileNotFoundException)

```

Comment:
>FileNotFoundException [](start = 18, length = 21)

Why not Anauthorizedaccessexcption?

Reply:
In this case, we want to avoid the race if the file is deleted after the file existence is check to avoid `FileNotFoundExcpetion` to be propagated to the client. `UnauthorizedException` is a different type of issue and it should be handled by the client separately.